### PR TITLE
Update to surfman 0.2

### PR DESCRIPTION
**Pull request by asajeffrey**
This is the surfman 0.2 upgrade for webxr.

### CODE REVIEW

**Comment by asajeffrey** [comment]
It shouldn't land until at least https://github.com/servo/surfman/pull/157 does.

File: webxr/openxr/mod.rs [hunk 1]
```diff
@@ -288,7 +283,7 @@ impl SurfaceProvider for OpenXrProvider {
                 device.create_surface_from_texture(
                     context,
                     &Size2D::new(size.width, size.height),
-                    self.images[image as usize],
+                    self.images[image as usize] as _,
```
Comment (jdm) [inline webxr/openxr/mod.rs]: What's the cast required for?

File: webxr/openxr/mod.rs [hunk 2]
```diff
@@ -371,17 +365,14 @@ impl OpenXrDevice {
...
-        let _ = gl_thread.execute(Box::new(move || {
+        let _ = gl_thread.execute(Box::new(move |device| {
             // Get the current surfman device and extract it's D3D device. This will ensure
             // that the OpenXR runtime's texture will be shareable with surfman's surfaces.
-            let (device, mut context) = unsafe {
-                SurfmanDevice::from_current_context().expect("Failed to create graphics context!")
-            };
-            device.destroy_context(&mut context).unwrap();
-            let d3d_device = device.d3d11_device();
+	    let native_device = device.native_device();
```
Comment (jdm) [inline webxr/openxr/mod.rs]: nit: indentation

File: webxr/openxr/mod.rs (hunk 1 above)
Comment (asajeffrey) [reply webxr/openxr/mod.rs]: Oops, it isn't.

File: webxr/openxr/mod.rs (hunk 2 above)
Comment (asajeffrey) [reply webxr/openxr/mod.rs]: Fixed.

**Comment by asajeffrey** [comment]
@bosrs-servo r=manishearth

**Comment by asajeffrey** [comment]
Can't spell... @bors-servo r=Manishearth

**Pull request merged**